lib/osut/utils.rb

```diff
@@ -31,20 +31,26 @@
 require "openstudio"
 
 module OSut
-  # DEBUG for devs; WARN/ERROR for users (bad OS input), see OSlg
   extend OSlg
 
-  TOL  = 0.01             # default distance tolerance (m)
-  TOL2 = TOL * TOL        # default area tolerance (m2)
-  DBG  = OSlg::DEBUG.dup  # see github.com/rd2/oslg
-  INF  = OSlg::INFO.dup   # see github.com/rd2/oslg
-  WRN  = OSlg::WARN.dup   # see github.com/rd2/oslg
-  ERR  = OSlg::ERROR.dup  # see github.com/rd2/oslg
-  FTL  = OSlg::FATAL.dup  # see github.com/rd2/oslg
-  NS   = "nameString"     # OpenStudio object identifier method
-
-  HEAD = 2.032 # standard 80" door
-  SILL = 0.762 # standard 30" window sill
+  DBG  = OSlg::DEBUG.dup # see github.com/rd2/oslg
+  INF  = OSlg::INFO.dup  # see github.com/rd2/oslg
+  WRN  = OSlg::WARN.dup  # see github.com/rd2/oslg
+  ERR  = OSlg::ERROR.dup # see github.com/rd2/oslg
+  FTL  = OSlg::FATAL.dup # see github.com/rd2/oslg
+  NS   = "nameString"    # OpenStudio object identifier method
+  TOL  = 0.01            # default distance tolerance (m)
+  TOL2 = TOL * TOL       # default area tolerance (m2)
+  HEAD = 2.032           # standard 80" door
+  SILL = 0.762           # standard 30" window sill
+  DMIN = 0.010           # min. material thickness
+  DMAX = 1.000           # max. material thickness
+  KMIN = 0.010           # min. material thermal conductivity
+  KMAX = 2.000           # max. material thermal conductivity
+  UMAX = KMAX / DMIN     # material USi upper limit, 200.000
+  UMIN = KMIN / DMAX     # material USi lower limit,   0.010
+  RMIN = 1.0 / UMAX      # material RSi lower limit,   0.005 (or R-IP   0.03)
+  RMAX = 1.0 / UMIN      # material RSi upper limit, 100.000 (or R-IP 567.80)
```

Comment:
Extending _OSut_ constants to cover material MIN/MAX thresholds, e.g.:

- thickness
- thermal conductivity
- thermal resistance
- thermal conductance

The goal is to ensure that TBD and BTAP inherit the very same constants, preventing conflicts with extreme cases (e.g. when TBD _uprates_ a construction).